### PR TITLE
Proxy/config loader refactor

### PR DESCRIPTION
**Pull request by kubistika**
## Regarding configuration

- Config loading code is cleaner.
- Only open allowed channels (do not allow disp if DisplayControl is not allowed in config).

## Regarding filters
- Make sure that `filter_init` is exported on filters compilation.
- Instead of dropping the connection in filter_demo, just print some info to stdout.

### CODE REVIEW

File: server/proxy/config.ini [hunk 1]
```diff
@@ -18,19 +18,14 @@ Port = 3389
 Mouse = 1
 Keyboard = 1
 
-[Graphics]
-GFX = 1
-BitmapUpdate = 1
```
Comment (speidy) [inline server/proxy/config.ini]: we don't want that one anymore?

File: server/proxy/pf_config.c [hunk 2]
```diff
@@ -30,195 +30,143 @@
...
 #define TAG PROXY_TAG("config")
 
-#define CHANNELS_SEPERATOR ","
+static inline BOOL pf_config_get_bool(wIniFile* ini, const char* section, const char* key)
+{
+	return IniFile_GetKeyValueInt(ini, section, key);
+}
 
-wArrayList* parse_string_array_from_str(const char* str)
+static BOOL pf_config_get_uint16(wIniFile* ini, const char* section, const char* key, UINT16* port)
```
Comment (speidy) [inline server/proxy/pf_config.c]: port -> value

File: server/proxy/config.ini (hunk 1 above)
Comment (kubistika) [reply server/proxy/config.ini]: nope, gfx moved to channels section and we never used BitmapUpdate value

File: server/proxy/pf_config.c (hunk 2 above)
Comment (kubistika) [reply server/proxy/pf_config.c]: just a second after you comment, renamed to result

File: server/proxy/filters/filters_api.h [hunk 3]
```diff
@@ -22,6 +22,20 @@
 #ifndef FREERDP_SERVER_PROXY_FILTERS_API_H
 #define FREERDP_SERVER_PROXY_FILTERS_API_H
 
+#if defined _WIN32 || defined __CYGWIN__
```
Comment (hardening) [inline server/proxy/filters/filters_api.h]: IIRC we have something equivalent somewhere else, can you check that  ?

**Comment by hardening** [comment]
@freerdp-bot can you test ?

**Comment by hardening** [comment]
side note: VS2010 shows some interesting warnings that may/should be fixed

File: server/proxy/filters/filters_api.h (hunk 3 above)
Comment (kubistika) [reply server/proxy/filters/filters_api.h]: We have FREERDP_API/WINPR_API
Maybe i can #define PROXY_API FREERDP_API

**Comment by kubistika** [comment]
@hardening changed the definition of PROXY_API and changed the inline function to be a macro, that should fix the msbuild errors

could you test again?

File: server/proxy/filters/filters_api.h (hunk 3 above)
Comment (hardening) [reply server/proxy/filters/filters_api.h]: sounds nice

Comment (kubistika) [reply server/proxy/filters/filters_api.h]: ok please test again

**Comment by hardening** [comment]
@freerdp-bot can you test ?

**Comment by kubistika** [comment]
@hardening can we merge this one?

**Pull request merged**